To-Do-app/src/features/userSlice.ts

```diff
@@ -0,0 +1,30 @@
+import { createSlice } from "@reduxjs/toolkit"
+import { RootState } from "../app/store"
+
+export const userSlice = createSlice({
+    name: "user",
+    initialState: { user: null, username: null, loggedIn: false },
+    reducers: {
+        login: (state, action) => {
+            state.user = action.payload
+            state.username = action.payload.email
+            state.loggedIn = true
+        },
+
+        logout: (state, action) => {
+            state.loggedIn = false
```

Comment:
when you log out, you still will have the user details in the store ?

Reply:
Yes sir, The user details will be stored and loggedIn status will be set to false after logout

Reply:
What is the use of capturing the email when some one has already logged out ?

Reply:
To keep a track of the correct user